core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -1029,21 +1030,54 @@ static void loadDirectoryMetadata(RpcContext rpcContext, LockedInodePath inodePa
     if (inodePath.fullPathExists()) {
       return;
     }
+    MountTable.Resolution resolution = mountTable.resolve(inodePath.getUri());
+    // create the actual metadata
+    loadDirectoryMetadataInternal(rpcContext, context, inodePath, resolution.getUri(),
+        resolution.getUfsClient(), fsMaster, resolution.getShared(),
+        mountTable.isMountPoint(inodePath.getUri()));
+  }
+
+  /**
+   * Loads metadata for the directory that will be the mount point.
+   *
+   * This difference between this method and
+   * {@link InodeSyncStream#loadDirectoryMetadata(RpcContext, LockedInodePath, LoadMetadataContext,
+   * MountTable, DefaultFileSystemMaster)} is that it accepts more specific parameters(i.e
+   * . isShared, ufsUri and ufsClient) and doesn't require the MountTable as its parameter.
+   */
+  static void loadMountPointDirectoryMetadata(RpcContext rpcContext, LockedInodePath inodePath,
+      LoadMetadataContext context, boolean isShared, AlluxioURI ufsUri,
+      UfsManager.UfsClient ufsClient, DefaultFileSystemMaster fsMaster) throws
+      FileDoesNotExistException, IOException, InvalidPathException, AccessControlException {
+    if (inodePath.fullPathExists()) {
+      return;
+    }
```

Comment:
What does it mean if the inode already exists? Should you throw an error instead?

Reply:
The origininal `loadDirectoryMetadata` has this `inodePath` check and returns immediately with no error threw if `fullPathExists` is true. So I think it would be better if we keep this logic intact?

Reply:
i checked the code for this part and the logic is:
1. The sync will `return` if the inode path is there, because all that the sync cares is to finish the sync
2. The `DefaultFileSystemMaster.mountInternal()` will throw an exception if the inode path is there

Therefore let's keep the code as-is here as you mentioned, but add the comments below:
```
// Return if the full path exists because the sync cares only about keeping up with the UFS
// If the mount point target path exists, the later mount logic will complain
```